sql/core/src/test/scala/org/apache/spark/sql/execution/ShufflePartitionsUtilSuite.scala

```diff
@@ -459,9 +466,36 @@ class ShufflePartitionsUtilSuite extends SparkFunSuite {
         Seq(
           Some(specs1),
           Some(specs2)),
-        targetSize, 1)
+        targetSize)
       assert(coalesced == Seq(expected1, expected2))
     }
+
+    {
+      // No actual coalescing happened, return empty result.
```

Comment:
not related to this PR. Just found we missed testing this case when reading the code.